src/sagemaker/workflow/pipeline.py

```diff
@@ -37,48 +37,58 @@
 from sagemaker.workflow.pipeline_experiment_config import PipelineExperimentConfig
 from sagemaker.workflow.parallelism_config import ParallelismConfiguration
 from sagemaker.workflow.properties import Properties
-from sagemaker.workflow.steps import Step
+from sagemaker.workflow.steps import Step, StepTypeEnum
 from sagemaker.workflow.step_collections import StepCollection
 from sagemaker.workflow.condition_step import ConditionStep
 from sagemaker.workflow.utilities import list_to_request
 
+_DEFAULT_EXPERIMENT_CFG = PipelineExperimentConfig(
+    ExecutionVariables.PIPELINE_NAME, ExecutionVariables.PIPELINE_EXECUTION_ID
+)
+
 
-@attr.s
```

Comment:
Removing the `@attr.s` and define `__init__` directly, as the former is not thread safe. 
With the use of `@attr.s`, multiple unit tests of the PR failed due to `ValueError("Pipeline steps cannot have duplicate names.")`, when invoking `_generate_step_map(self.steps, self._step_map)`
The reason is, in this case, in multiple unit tests, `_generate_step_map` methods are building the same global `_step_map`. As a result, if a step name duplicates in different tests, it can cause the `ValueError("Pipeline steps cannot have duplicate names.")`.